tests/OPT/failing_tests/gemm_i32.mlir

```diff
@@ -0,0 +1,167 @@
+// RUN: iree-compile --iree-hal-target-backends=amd-aie --compile-to=executable-sources %s | iree-opt --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-hal-translate-target-executable-variants{target=amd-aie})))" --split-input-file | FileCheck %s --check-prefix=CPP
+
+// This lit test keeps track of all GEMMs as seen in OPT.
+// The intention is to gradually retire the tests from this file by getting rid of `XFAIL`
```

Comment:
Want to be more clear on the goal of this test. Is this test used to track all the failed dispatches in OPT instead of all GEMMs?

Reply:
yes.. these are the sample programs we should be able to compile with IREE on amd-aie backend. 

THanks @Abhishek-Varma . This is a neat idea!